guacamole-ext/src/main/java/org/apache/guacamole/properties/CipherGuacamoleProperty.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.guacamole.properties;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.io.IOException;
+import java.lang.IllegalArgumentException;
+import java.security.InvalidKeyException;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.KeySpec;
+import java.security.spec.PKCS8EncodedKeySpec;
+import javax.crypto.Cipher;
+import javax.crypto.NoSuchPaddingException;
+import org.apache.guacamole.GuacamoleException;
+import org.apache.guacamole.environment.Environment;
+import org.apache.guacamole.environment.LocalEnvironment;
+
+/**
+ * A GuacamoleProperty whose value is derived from a private key file.
+ */
+public abstract class CipherGuacamoleProperty implements GuacamoleProperty<Cipher>  {
+
+    @Override
+    public Cipher parseValue(String value) throws GuacamoleException {
+
+        try {
+
+            final Environment environment = new LocalEnvironment();
+
```

Comment:
This feels like reverting back to behavior advised against before, but I'm not sure the best way to get the Guacamole Home directory into this property to open up the file without doing this.  Suggestions?

Reply:
Hm ... I see what you mean.

It would be possible to get Guice to inject the `Environment` by using a `Provider<CipherGuacamoleProperty>`, thus avoiding recreating things, but perhaps there is a different way.

This may end up reverting things back to how you had them, but what about relying on the value of this property to be the absolute path to the key? If the property itself ends up needing to be simply a `File` or `String`, with the actual reading into `PrivateKey` and finally `Cipher` being abstracted only within the configuration service, that's fine, I'm sure.

Reply:
Absolute path is fine - I think there are a few other similar properties (basic user mapping, even) that do that, so it isn't without precedent.  I guess it just seems like assuming it'll be in GUACAMOLE_HOME feels a little cleaner from a configuration perspective - admins don't have to keep guessing which properties assume absolutely paths and which are relative to GUACAMOLE_HOME - but as long as it's properly documented it should be okay.

Reply:
Pulled out home env call and we'll just require absolute path.